onboarding_auth/onboardingAuthDomain/src/main/java/com/alexafit/onboardingauthdomain/useCase/ValidateEmail.kt

```diff
@@ -1,19 +1,25 @@
 package com.alexafit.onboardingauthdomain.useCase
 
-class ValidateEmail {
+import com.alexafit.onboardingauthdomain.model.domain.mapper.ValidateEmailResult
 
-    operator fun invoke(email: String): Pair<String, Boolean> {
-        return when (email.isBlank()) {
-            true -> Pair(email.trim(), false)
-            false -> {
-                val trimmedEmail = email.trimEnd()
-                trimmedEmail.isEmailValid()
-            }
+class ValidateEmail {
+    operator fun invoke(email: String): ValidateEmailResult {
+        return if (email.isBlank()) {
+            ValidateEmailResult(
+                emailResult = email.trim(),
+                validEmail = false
+            )
+        } else {
+            val trimmedEmail = email.trimEnd()
+            trimmedEmail.isEmailValid()
         }
     }
+}
 
-    private fun String.isEmailValid(): Pair<String, Boolean> {
-        val isValid = android.util.Patterns.EMAIL_ADDRESS.matcher(this).matches()
-        return Pair(this, isValid)
-    }
+private fun String.isEmailValid(): ValidateEmailResult {
+    val isValid = android.util.Patterns.EMAIL_ADDRESS.matcher(this).matches()
+    return ValidateEmailResult(
+        emailResult = this,
+        validEmail = isValid
+    )
```

Comment:
Here, you're using an Android-specific dependency (as you can see in the imports) which doesn't really belong into domain. Instead, I'd recommend to create an abstraction for this to keep this class locally unit testable.

Every time, you have implementation details in domain and want to get rid of that, the procedure is the same:

1. You create an abstraction (interface) in domain
2. You write an implementation of that in data (in this case using Patterns.EMAIL_ADDRESS...)
3. You pass the abstraction to the constructor of your class needing it in domain
4. Now you can decide which implementation you pass when you create that class/use case
For example:

*In domain*
```
interface EmailPatternValidator {
    fun isValidEmailPattern(email: String): Boolean
}
```

*In data*
```
class EmailPatternValidatorImpl: EmailPatternValidator {

    override fun isValidEmailPattern(email: String): Boolean {
        return Patterns.EMAIL_ADDRESS.matcher(email).matches()
    }
}
```

Then inject an `EmailPatternValidator` into your use case and use there.